Add system::loadavg

vandrada opened the pull request:
Add function to return the system's load average. This includes the
average number of runnable entities averaged over one, five, and fifteen
minutes as well as statistics on the total number of runnable entities
and the last pid that was used.

Behind the scenes, the `try_parse` macro was moved to utils since it's
now used in multiple places.

rkday commented on src/system.rs:
```diff
@@ -122,6 +165,33 @@ fn uptime_internal(data: &str) -> isize {
+pub fn loadavg() -> Result<LoadAverage> {
+    let data = read_file(Path::new("/proc/loadavg"))?;
+    // strips off any trailing new line as well
+    let fields: Vec<&str> = data.split_whitespace().collect();
```
Could you split out a separate function from this point on, with the signature `fn loadavg_internal(proc_file_contents: &str) -> Result<LoadAverage>`?

That makes this much easier to test - you can call `loadavg_internal` and pass it some sample data and assert specific things about the results, rather than it parsing the system's /proc/uptime and you only being able to assert that it's greater than zero.

rkday commented on src/system.rs:
```diff
@@ -105,6 +107,47 @@ impl SwapMemory {
+}
+
+impl LoadAverage {
+    fn new(
```
Since `fn new` isn't doing any calculations or deriving any values, probably better to not have it and just use a struct literal (as is done for Fd at https://github.com/borntyping/rust-psutil/blob/master/src/process.rs#L568).

vandrada commented:
Went ahead and removed the whole `impl LoadAverage` block since all it had was the `new` and moved the actual parsing into an internal, testable function.

rkday commented:
@borntyping , any comments or am I good to merge?

borntyping commented:
Looks great to me!

borntyping commented:
(Oops, clicked something I didn't mean to.)
